src/core/imap/MCIMAPSession.cpp

```diff
@@ -2760,6 +2760,16 @@ IndexSet * IMAPSession::search(String * folder, IMAPSearchKind kind, String * se
             expr = IMAPSearchExpression::searchContent(searchString);
             break;
         }
+        case IMAPSearchKindRead:
```

Comment:
If you're adding those, could you also add the other search types that are flags-like?
- flagged
- answered
- draft
- deleted
- spam